falcon/util/misc.py

```diff
@@ -357,6 +358,8 @@ def secure_filename(filename: str) -> str:
     Args:
         filename (str): Arbitrary filename input from the request, such as a
             multipart form filename field.
+        max_length (int, optional): Maximum length of the sanitized filename,
```

Comment:
This `(int, optional)` looks a bit unconventional. I think you can simply remove this type from here altogether, and let Sphinx pick it up from the type annotation.

Reply:
Do you want to pass the max_length as a parameter in the **BodyPart.secure_filename** function or set it as an instance or class variable and reference it from there?

Reply:
I want us to actually use the new option in `BodyPart.secure_filename()`, i.e.:
```python
    @property
    def secure_filename(self) -> str:
        """The sanitized version of `filename` using only the most common ASCII
        <...>
        """  # noqa: D205
        try:
            return misc.secure_filename(self.filename or '')
        except ValueError as ex:
            raise MultipartParseError(description=str(ex)) from ex
```
:arrow_down:
```python
    @property
    def secure_filename(self) -> str:
        """The sanitized version of `filename` using only the most common ASCII
        <...>
        """  # noqa: D205
        try:
            return misc.secure_filename(self.filename or '', max_length=self._parse_options.max_secure_filename_length)
        except ValueError as ex:
            raise MultipartParseError(description=str(ex)) from ex
```
(untested)